### PR TITLE
Add option to trust all URLs to the same website

### PR DESCRIPTION
**Pull request by nschnitzer**
### Contains

Adds option to trust all URLs with a certain hostname.

### How to test

Try and open URLs with the same hostname after selecting the option.

### Outstanding before merging

If anything. You can use neat checkboxes! Feel free to delete if not needed

- [ ] Need to consider use case x
- [ ] Still have to adjust the wiki doc
- [ ] Will need translation work

### CODE REVIEW

**Comment by GooeyHub** [comment]
Can one of the admins please verify this patch?

File: engine/src/main/resources/assets/ui/menu/confirmUrlPopup.ui [hunk 1]
```diff
@@ -57,6 +57,23 @@
...
+                            "column-widths":[0.1,0.90],
+                            "contents": [
+                                {
+                                    "type": "UICheckbox",
+                                    "id": "saveHostName",
+                                    "family": "option-grid"
+                                },
+                                {
+                                    "type": "UILabel",
+                                    "text": "Always accept this urls with this hostname",
```
Comment (llvieira) [inline engine/src/main/resources/assets/ui/menu/confirmUrlPopup.ui]: `Always accept urls to this website` is better.

Comment (llvieira) [inline engine/src/main/resources/assets/ui/menu/confirmUrlPopup.ui]: `Always accept urls to this website` is better.

File: engine/src/main/java/org/terasology/config/WebBrowserConfig.java [hunk 2]
```diff
@@ -41,4 +42,21 @@ public boolean isUrlTrusted(String url) {
...
+    public void removeTrustedHostName(String hostname) {
+    	trustedHostNames.remove(hostname);
+    }
+    
+    public boolean isHostNameTrusted(String hostname) {
+    	return trustedHostNames.contains(hostname);
+    }
+    
+    public List<String> getTrustedHostNames()
+    {
```
[the remark this answers is not in the document]
Comment (jellysnake) [reply engine/src/main/java/org/terasology/config/WebBrowserConfig.java]: To add to this, if you are using intelliJ, then the in built formatting option (`ctrl+alt+L` on windows I believe) will format your code according to our rules

File: engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/ConfirmUrlPopup.java [hunk 3]
```diff
@@ -73,7 +76,23 @@ public void setMessage(String title, String message) {
...
+                if (isTrustedHostName) {
+                    webBrowserConfig.addTrustedHostName(hostName);
+                } else {
+                    webBrowserConfig.removeTrustedHostName(hostName);
+                }
+            });
+            
         UICheckbox saveUrl = find("saveUrl", UICheckbox.class);
 
         if (saveUrl != null && webBrowserConfig != null) {
```
Comment (jellysnake) [inline engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/ConfirmUrlPopup.java]: I don't think this is right. You've got this if statement inside the prior on that checks for the `saveHostName`.  

They should either both be seperate or they should combined into one statement.  
I would recommend seperate.

File: engine/src/main/java/org/terasology/rendering/nui/widgets/UIButtonWebBrowser.java [hunk 4]
```diff
@@ -30,144 +30,162 @@
...
-
-    public UIButtonWebBrowser setNuiManager(final NUIManager nuiManager) {
-        this.nuiManager = nuiManager;
-        return this;
-    }
-
-    public UIButtonWebBrowser setTranslationSystem(final TranslationSystem translationSystem) {
-        this.translationSystem = translationSystem;
-        return this;
-    }
```
Comment (jellysnake) [inline engine/src/main/java/org/terasology/rendering/nui/widgets/UIButtonWebBrowser.java]: Once again, you've intented this whole file and I don't think it was supposed to be. Try running the autoformat on the files before you commit them

File: engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/ConfirmUrlPopup.java (hunk 3 above)
Comment (nschnitzer) [reply engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/ConfirmUrlPopup.java]: This was not my implementation. This previously existed in the repository.

Comment (nschnitzer) [reply engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/ConfirmUrlPopup.java]: Also, Java ignores white spaces. It's just a styling tweak.

File: engine/src/main/java/org/terasology/config/WebBrowserConfig.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (nschnitzer) [reply engine/src/main/java/org/terasology/config/WebBrowserConfig.java]: I am using eclipse

[the remark this answers is not in the document]
Comment (llvieira) [reply engine/src/main/java/org/terasology/config/WebBrowserConfig.java]: In Eclipse, to format your code you can press `Ctrl + Shift + F`. If it doesn't work, I would suggest you to install IntelliJ and fix the indent issue.

**Comment by Cervator** [comment]
This pull request **introduces 1 alert** when merging c4a26798cf38a578a547cabd54d9074ee99a8955 into 9b83c0150b9a6c4b53809c381449ac2a031881d2 - [view on LGTM.com](https://lgtm.com/projects/g/MovingBlocks/Terasology/rev/pr-cd6f70a68945cca7543b05e1785643f3dcd8c02b)

**new alerts:**

* 1 for Useless null check

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by jellysnake** [comment]
Student is using Eclipse which has different formatting rules to intelliJ. Whoever merges this could they please re-format all affected files in intelliJ to fix indent issues :)

(Also would be nice if they fixed up the issue with the `if`s being nested when they should be seperate. That's just as simple as moving a `}` around.)

**Comment by nschnitzer** [comment]
The files need to be formatted in IntelliJ before being merged.

**Comment by GooeyHub** [comment]
Can one of the admins please verify this patch?

**Pull request closed without merging**